src/lib/flow/sol-flow-parser.c

```diff
@@ -409,8 +423,47 @@ create_fbp_type(
     return 0;
 }
 
+#ifdef JAVASCRIPT
+static int
+create_js_type(
+    struct parse_state *state,
+    struct sol_str_slice name,
+    struct sol_str_slice contents,
+    const struct sol_flow_node_type **type)
+{
+    const struct sol_flow_parser_client *client = state->parser->client;
+    const char *buf, *filename;
+    struct sol_flow_node_type *result;
+    size_t size;
+    int err;
+
+    SOL_NULL_CHECK(client, -ENOSYS);
+    SOL_NULL_CHECK(client->read_file, -ENOSYS);
+
+    filename = sol_arena_strdup_slice(state->arena, contents);
+    err = client->read_file(client->data, filename, &buf, &size);
+    if (err < 0)
+        return -EINVAL;
+
+    result = sol_flow_js_new_type(buf, size);
+    if (!result)
```

Comment:
No, the ownership of buf belongs to those who implemented client->read_file(), and there is a promise that it must live until the flow-parser API call retuns.